test/ha/activator_test.go

```diff
@@ -0,0 +1,101 @@
+// +build e2e
+
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ha
+
+import (
+	"log"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/serving/pkg/apis/autoscaling"
+	revisionresourcenames "knative.dev/serving/pkg/reconciler/revision/resources/names"
+	rtesting "knative.dev/serving/pkg/testing/v1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+)
+
+const (
+	activatorDeploymentName = "activator"
+	activatorLabel          = "app=activator"
+)
+
+// Activator is managed by the activator HPA and does not have leader election enabled.
+// The test ensures that stopping one of the activator pods doesn't affect user applications.
+func TestActivatorHA(t *testing.T) {
+	clients := e2e.Setup(t)
+
+	names, resources := createPizzaPlanetService(t,
```

Comment:
A bit more documentation would be great. I.e. if I read the test correctly you're creating two services, one to continuously spoof and one to check if you can scale it back up from 1?

Reply:
Yeah. You understand it correctly. I can add more comments yeah